### PR TITLE
[DOCS] Add token graph concept docs

### PR DESCRIPTION
**Pull request by jrodewig**
Adds conceptual docs for token graphs.
These docs cover:

* How a token graph is constructed from a token stream
* How synonyms and multi-position tokens impact token graphs
* How token graphs are used during search
* Why some token filters produce invalid token graphs

Also makes the following supporting changes:
* Adds anchors to the [Anatomy of an Analyzer][0] docs for cross-linking
* Adds several SVGs for token graph diagrams

### Preview link
http://elasticsearch_53339.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/token-graphs.html

[0]: https://www.elastic.co/guide/en/elasticsearch/reference/master/analyzer-anatomy.html

### CODE REVIEW

File: docs/reference/analysis/token-graphs.asciidoc [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+image::images/analysis/token-graph-dns-synonym-ex.svg[align="center"]
+
+[[token-graphs-token-graphs-search]]
+===== Using token graphs for search 
+
+<<analysis-index-search-time,Indexing>> ignores the `positionLength` attribute
+and does not support token graphs containing multi-position tokens.
+
+However, queries, such as the <<query-dsl-match-query-phrase,`match_phrase`>>
```
Comment (jimczi) [inline docs/reference/analysis/token-graphs.asciidoc]: `match` queries are also affected. The input `ny city` with a `ny, new york` expansion creates an incorrect `(ny OR new) AND york AND city` query with the `synonym` filter.

**Review by jimczi — APPROVED** [review]
❤️ @jrodewig 
I left one comment, we could also add a link to this new section in the `synonym_graph`  and `word-delimiter-graph` docs ?

File: docs/reference/analysis/token-graphs.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/analysis/token-graphs.asciidoc]: I updated this with d4a711d6154 to mention the `match` query.

**Comment by jrodewig** [comment]
Thanks @jimczi. I've added some cross-links to this page in the `synonym_graph` and `word-delimiter-graph` docs.

**Comment by jrodewig** [comment]
### Backport commits
master 8d5478f
7.x 8f4a3eb07fb
7.6 f27c1883b15

**Pull request merged**